src/seat/mod.rs

```diff
@@ -354,26 +375,6 @@ impl<D> RegistryHandler<D> for SeatState
 where
     D: Dispatch<wl_seat::WlSeat, SeatData> + SeatHandler + ProvidesRegistryState + 'static,
 {
-    fn ready(state: &mut D, conn: &Connection, qh: &QueueHandle<D>) {
-        let seats = state
-            .registry()
-            .bind_all(qh, 1..=7, |id| SeatData {
-                has_keyboard: Arc::new(AtomicBool::new(false)),
-                has_pointer: Arc::new(AtomicBool::new(false)),
-                has_touch: Arc::new(AtomicBool::new(false)),
-                name: Arc::new(Mutex::new(None)),
-                id,
-            })
-            .expect("failed to bind global");
-
-        for seat in seats {
-            let data = seat.data::<SeatData>().unwrap().clone();
-
-            state.seat_state().seats.push(SeatInner { seat: seat.clone(), data });
-            state.new_seat(conn, qh, seat);
```

Comment:
When we create the seat state, there is no notification initially that any new seats were created. Or should it be expected that the user would iterate over the initially available seats?

Reply:
> Or should it be expected that the user would iterate over the initially available seats?

This doesn't seem unreasonable since it's the same as how the use would be expected to handle globals, if they used `GlobalState` to iterate over globals available initially and detect when any are added/removed.

Otherwise, it runs into the problem of how to defer calling that callback until the state is available.